iothub/service/src/Registry2/ConfigurationsClient.cs

```diff
@@ -0,0 +1,341 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Common.Exceptions;
+using Microsoft.Azure.Devices.Http2;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Subclient of <see cref="IotHubServiceClient"/> that handles creating, updating, getting and deleting configurations.
+    /// </summary>
+    public class ConfigurationsClient
+    {
+        private string _hostName;
+        private IotHubConnectionProperties _credentialProvider;
+        private HttpClient _httpClient;
+        private HttpRequestMessageFactory _httpRequestMessageFactory;
+
+        private const string ConfigurationRequestUriFormat = "/configurations/{0}";
+        private const string ConfigurationsRequestUriFormat = "&top={0}";
+        private static readonly TimeSpan s_defaultOperationTimeout = TimeSpan.FromSeconds(100);
+
+        /// <summary>
+        /// Creates an instance of this class. Provided for unit testing purposes only.
+        /// </summary>
+        protected ConfigurationsClient()
+        {
+        }
+
+        internal ConfigurationsClient(string hostName, IotHubConnectionProperties credentialProvider, HttpClient httpClient, HttpRequestMessageFactory httpRequestMessageFactory)
+        {
+            _credentialProvider = credentialProvider;
+            _hostName = hostName;
+            _httpClient = httpClient;
+            _httpRequestMessageFactory = httpRequestMessageFactory;
+        }
+
+        /// <summary>
+        /// Creates a new configuration for Azure IoT Edge in IoT hub
+        /// </summary>
+        /// <param name="configuration">The configuration object being created.</param>
+        /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
+        /// <returns>The Configuration object.</returns>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public async virtual Task<Configuration> CreateAsync(Configuration configuration, CancellationToken cancellationToken = default)
```

Comment:
Need to add parameter validation for ```configuration``` before you use it. See [here](https://github.com/Azure/azure-iot-sdk-csharp/blob/4164acb88e987cfbf92ed3b97c8fd14c17dc6a06/iothub/service/src/Registry2/DevicesClient.cs#L76) for an example